File: Leetcode/Easy Problems/1. Two Sum/dec-20-2023.cpp
/*
  December 20, 2023
  - Runtime Achieved: 3 ms
  - Memory Achieved: 11.5 MB
*/
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> output;
        int index = 0;
        for(int x: nums) {
            int y = target - x;
            if(!(output.find(y) == output.end()))
                return {output[y], index};
            output[x] = index;
            index++;
        } 
        return {-1,-1};
    }
};
  

File: Leetcode/Medium Problems/3. Longest Substring Without Repeating Characters/dec-20-2023.py
# December 20, 2023
# - Runtime Achieved: 55 ms 
# - Memory Achieved: 16.5 MB
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        temp = {}
        ln = len(s)
        n, ll = 0, 0
        for lr in range(ln): # runs through str to lr
            uchar = s[lr] # new character stored
            if uchar in temp and ll <= temp[uchar]:
                ll = 1 + temp[uchar] # adds count of mutiples
            else:
                n = max(n,lr-ll+1) # finds max count from ll multiples minus str length + 1
            temp[uchar] = lr
        return n
